packages/realm-react/src/__tests__/RealmProvider.test.tsx

```diff
@@ -43,317 +44,556 @@ const catSchema: Realm.ObjectSchema = {
   },
 };
 
-const { RealmProvider, useRealm } = createRealmContext({
+const withConfigRealmContext = createRealmContext({
   schema: [dogSchema],
   inMemory: true,
   path: randomRealmPath(),
 });
 
-const EmptyRealmContext = createRealmContext();
-
 describe("RealmProvider", () => {
   afterEach(() => {
     Realm.clearTestState();
   });
 
-  it("returns the configured realm with useRealm and closes on unmount", async () => {
-    const wrapper = ({ children }: { children: React.ReactNode }) => <RealmProvider>{children}</RealmProvider>;
-    const { result, unmount } = renderHook(() => useRealm(), { wrapper });
-    await waitFor(() => expect(result.current).not.toBe(null));
-    const realm = result.current;
-    expect(realm).not.toBe(null);
-    expect(realm.schema[0].name).toBe("dog");
-    unmount();
-    expect(realm.isClosed).toBe(true);
-  });
+  describe("with a Realm Configuration", () => {
+    const { RealmProvider, useRealm } = withConfigRealmContext;
 
-  it("returns the configured realm with useRealm and stays open if flagged", async () => {
-    const wrapper = ({ children }: { children: React.ReactNode }) => (
-      <RealmProvider closeOnUnmount={false}>{children}</RealmProvider>
-    );
-    const { result, unmount } = renderHook(() => useRealm(), { wrapper });
-    await waitFor(() => expect(result.current).not.toBe(null));
-    const realm = result.current;
-    expect(realm.schema[0].name).toBe("dog");
-    unmount();
-    expect(realm.isClosed).toBe(false);
-  });
+    it("returns the configured realm with useRealm", async () => {
+      const wrapper = ({ children }: { children: React.ReactNode }) => <RealmProvider>{children}</RealmProvider>;
+      const { result } = renderHook(() => useRealm(), { wrapper });
+      await waitFor(() => expect(result.current).not.toBe(null));
```

Comment:
Can we update the `not.toBe(null)`s to the more specific `toBeInstanceOf(Realm)` instead? 🙂